### PR TITLE
Add pip check

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jan-janssen** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -32,6 +32,7 @@ requirements:
     - typing_extensions >=3.10.0.0
 #    - windows-curses  # [win]
     - wrapt
+    - msgpack-python
```
Comment (jan-janssen) [inline recipe/meta.yaml:35]: If the dependency is not added the pip check fails as shown in https://github.com/conda-forge/python-can-feedstock/pull/17

Comment (ocefpaf) [reply recipe/meta.yaml:35]: If `msgpack-python` is optional but ill defined in the package metadata as such, it is fine to skip pip-check. However, downstream packages should also drop it. Ideally we should fix the pkg metadata and declare it optional. However, if the upstream devs want to make it a hard requirement the solution here is good to merge.

**Comment by jan-janssen** [comment]
@danielhrisca and @hardbyte Can you review this pull request?

**Pull request closed without merging**